TheMET/TheMET/TagListView/TagListView.swift

```diff
@@ -0,0 +1,205 @@
+            configuration.contentInsets = NSDirectionalEdgeInsets(top: 5, leading: 10, bottom: 5, trailing: 10)
+            let newTagView = TagView(configuration: configuration)
+            newTagView.setTitle(tag, for: .normal)
+            newTagViews.append(newTagView)
+            newTagView.translatesAutoresizingMaskIntoConstraints = false
+            newTagView.frame = CGRect(origin: .zero,
+                                      size: newTagView.intrinsicContentSize)
+            newTagView.translatesAutoresizingMaskIntoConstraints = true
+            newTagView.autoresizingMask = [ .flexibleRightMargin, .flexibleBottomMargin ]
+        }
+        self.tagViews = newTagViews
+        self.layoutTagViews()
+    }
+    
+    private func layoutTagViews() {
+        let visibleTags:[CGRect] = self.calculateLayout(for: self.bounds.width).tagViewFrames
+        let visibleTagsCount = visibleTags.count
+    throughTags: for (tagViewIndex, tagView) in tagViews.enumerated() {
+        if tagViewIndex < visibleTagsCount {
+            self.addSubview(tagView)
+            tagView.apply(radius: 15, backgroundColor: UIColor(named: "plum"), fontColor: UIColor(named: "blackberry"), font: NSLocalizedString("san_serif_font", comment: ""), fontSize: 13, buttonTitle: (self.tagViews[tagViewIndex].titleLabel?.text)!)
+            let tagFrameOrigin = self.calculateLayout(for: self.bounds.width).tagViewFrames[tagViewIndex].origin
+            tagView.frame.origin = tagFrameOrigin
+        }
+    }
+        if let showMoreButtonFrame = self.calculateLayout(for: self.bounds.width).showMoreButtonFrame,
+           showMoreButtonFrame.width != 0 {
+            self.addSubview(self.showMoreTagsButton)
+            self.showMoreTagsButton.frame.origin = showMoreButtonFrame.origin
+        }
+        self.invalidateIntrinsicContentSize()
+    }
+    
+    private func tagsWidth(from index1: Int, to index2: Int, xSpace: CGFloat) -> CGFloat {
+        var result: CGFloat = 0
+        for index in index1 ... index2 {
+            let tagView = self.tagViews[index]
+            result += tagView.bounds.width
+        }
+        result += xSpace * CGFloat(index2 - index1) + self.tagViews[index1 - 1].bounds.width
+        return result
+    }
+    
+    private func setRestTagsFrames(from index1: Int, to index2: Int, xSpace: CGFloat, ySpace: CGFloat, x: CGFloat, y: CGFloat) -> [CGRect] {
+        var tagViewFrames: [CGRect] = []
+        var xCoord: CGFloat = x
+        for index in index1 ... index2 {
+            let tagView = self.tagViews[index]
+            tagViewFrames.append(CGRect(x: xCoord, y: y, width: tagView.bounds.width, height: tagView.bounds.height))
+            xCoord += tagView.bounds.width + xSpace
+        }
+        return tagViewFrames
+    }
+    
+    private func setRestTagsAndShowMoreButtonFrames(from index1: Int, to index2: Int, xSpace: CGFloat, ySpace: CGFloat, x: CGFloat, y: CGFloat) -> (tagViewFrames: [CGRect], showMoreButton: CGRect) {
```

Comment:
1. This method doesn't set tags and showButton frames anymore. Rename.
2. The result tuple is inconsistent. You use term `tagViewFrames` for multiple frames but `showMoreButton` for a single frame. I suggest to rename the last one to `showMoreButtonFrame`

Reply:
fixed